core/dyn-configs/src/lib.rs

```diff
@@ -0,0 +1,14 @@
+#![doc = include_str!("../README.md")]
+
+use once_cell::sync::Lazy;
+use std::sync::atomic::{AtomicU64, Ordering};
+
+pub static EXPECTED_SHUTDOWN_AT: Lazy<AtomicU64> = Lazy::new(|| AtomicU64::new(0));
```

Comment:
I think it should be sufficient to have a `Lazy<u64>` instead of an `Lazy<AtomicU64>`.
The value is already inside a `sync` container.
Maybe change it to `Lazy<Option<BlockHeight>>` for readability:
* `BlockHeight` instead of `u64`.
* `None` instead of a special value `0`.

Reply:
Hi @nikurt,
I am not sure it is a good idea to use `unsafe` for this, because a mutable static`u64` can be mutated by multiple threads.

```
    --> chain/client/src/client_actor.rs:1204:109
     |
1204 |             if let Some(Some(block_height_to_shutdown)) = once_cell::sync::Lazy::<Option<BlockHeight>>::get(&EXPECTED_SHUTDOWN_AT) {
     |                                                                                                             ^^^^^^^^^^^^^^^^^^^^^ use of mutable static
     |
     = note: mutable statics can be mutated by multiple threads: aliasing violations or data races will cause undefined behavior
```

The way we can read the mutable static `u64` is unsafe.
`if let Some(Some(block_height_to_shutdown)) = Lazy::<Option<BlockHeight>>::get(unsafe{&EXPECTED_SHUTDOWN_AT}) {` 

No matter whether we keep using the current version of once_cell(`1.13.1`) or using `force_mut` from the latest version(`1.16.0`).  We always need unsafe to modify the static `Option<u64>`

**1.13.1**
`**std::pin::Pin::new(unsafe{&mut EXPECTED_SHUTDOWN_AT}).get_mut() = expected_shutdown;`

**1.16.0**
`*Lazy::force_mut(unsafe {&mut EXPECTED_SHUTDOWN_AT}) = expected_shutdown;`

The issue may non-exist in some computer architecture, but not all.
To avoid any kind of unexpected shutdowns, could we keep using `AtomicU64`?  

If there is something misunderstood please kindly let me know.  Thank you.